src/Features/Core/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -507,29 +492,19 @@ public override void RemoveProject(ProjectId projectId)
 
                 // make sure we get state even when none of our analyzer has ran yet. 
                 // but this shouldn't create analyzer that doesnt belong to this project (language)
-                var state = _analyzersAndState.GetOrCreateDiagnosticState(stateType, providerId, provider, document.Project.Id, document.Project.Language);
-                if (state == null)
+                var state = stateSet.GetState(stateType);
+
+                // see whether we can use existing info
+                var existingData = await state.TryGetExistingDataAsync(document, cancellationToken).ConfigureAwait(false);
+                if (existingData != null && versionCheck(existingData.TextVersion, existingData.DataVersion))
                 {
-                    if (!requireUpToDateDocumentDiagnostic)
```

Comment:
Why was this requireUpToDateDocumentDiagnostic check removed? did it move somewhere else? diff view is not enabling me to figure out what happened here.